docs/pluggable-discovery-specification.md

```diff
@@ -377,3 +377,67 @@ myboard.upload_port.1.apples=40
 
 will match on both `pears=20, apples=30` and `pears=30, apples=40` but not `pears=20, apples=40`, in that sense each
 "set" of identification properties is independent from each other and cannot be mixed for port matching.
+
+#### Identification of board options
+
+The board identification can also identify [custom board options](platform-specification.md#custom-board-options) if the
+corresponding `upload_port` properties are defined:
+
+```
+BOARDNAME.menu.MENUNAME.MENUOPTION.upload_port.IDENTIFIER=Value
+BOARDNAME.menu.MENUNAME.MENUOPTION.upload_port.#.IDENTIFIER=Value
+```
```

Comment:
````suggestion
If only a single set of identification properties are associated with the option:

```
BOARD_ID.menu.MENU_ID.OPTION_ID.upload_port.PORT_PROPERTY_KEY=PORT_PROPERTY_VALUE
```

If one or more sets of identification properties are associated with the option, an index number is used for each set:

```
BOARD_ID.menu.MENU_ID.OPTION_ID.upload_port.SET_INDEX.PORT_PROPERTY_KEY=PORT_PROPERTY_VALUE
```
````

- Make it clear there are two alternative formats.
- Use the same placeholders as in the custom board options specification.
- Use standard UPPER_SNAKE_CASE format for all placeholders.
- Make placeholders more descriptive.